### PR TITLE
new package(s): py-torch-geometric

### PR DESCRIPTION
**Pull request by aweits**
(with related dependencies: py-rdflib, py-googledrivedownloader)

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-rdflib/package.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    implementations store implementations for in-memory storage and
+    persistent storage on top of the Berkeley DB a SPARQL 1.1
+    implementation - supporting SPARQL 1.1 Queries and Update
+    statements """
+
+    homepage = "https://github.com/RDFLib/rdflib"
+    url      = "https://pypi.io/packages/source/r/rdflib/rdflib-5.0.0.tar.gz"
+
+    version('5.0.0', sha256='78149dd49d385efec3b3adfbd61c87afaf1281c30d3fcaf1b323b34f603fb155')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rdflib/package.py:22]: Missing setuptools dependency

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    small and single giant graphs, multi gpu-support, a large number
+    of common benchmark datasets (based on simple interfaces to create
+    your own), and helpful transforms, both for learning on arbitrary
+    graphs as well as on 3D meshes or point clouds."""
+
+    homepage = "https://github.com/rusty1s/pytorch_geometric"
+    url      = "https://github.com/rusty1s/pytorch_geometric/archive/1.6.0.tar.gz"
+
+    version('1.6.0', sha256='7d5231cdcc2ebd4444f406cbf1537eb49bf90ab6f446eaf1b7af5cdbe105f3c9')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-geometric/package.py:22]: Missing setuptools dependency

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    depends_on('py-tqdm', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-networkx', type=('build', 'run'))
+    depends_on('py-scikit-learn', type=('build', 'run'))
+    depends_on('py-numba', type=('build', 'run'))
+    depends_on('py-requests', type=('build', 'run'))
+    depends_on('py-pandas', type=('build', 'run'))
+    depends_on('py-rdflib', type=('build', 'run'))
+    depends_on('py-googledrivedownloader', type=('build', 'run'))
+    depends_on('py-h5py~mpi', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-torch-geometric/package.py:37]: I don't see this dependency listed

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:22]: Also, depends on Python 3.6+

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:22]: Can you check the build log and see if it uses `pytest-runner` or automatically downloads and installs it? I'm always wary of `setup_requires` dependencies, so I usually add them as build deps.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py (hunk 3 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:37]: I've taken to shutting off mpi support in hdf5-related stuff when it comes in contact with torch, as when users engage the parallel dataloader (which uses fork()) mpi tends to want to crash in nonobvious ways - I think I'm also seeing this in tensorflow now as well...

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py (hunk 2 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:22]: It wasn't pulling it down, but added it as a dep anyways.

File: var/spack/repos/builtin/packages/py-torch-geometric/package.py (hunk 3 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:37]: But does this package even depend on `py-h5py`?

Comment (aweits) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:37]: yes...?
```
coltrane (Downloads/pytorch_geometric-1.6.0):grep -r "h5py" *
docker/Dockerfile:RUN conda install -y h5py=2.8.0 \
docker/Dockerfile:RUN pip install h5py-cache==1.0
setup.py:    'h5py',
torch_geometric/datasets/dynamic_faust.py:import h5py
torch_geometric/datasets/dynamic_faust.py:        fm = h5py.File(self.raw_paths[0], 'r')
torch_geometric/datasets/dynamic_faust.py:        ff = h5py.File(self.raw_paths[1], 'r')
torch_geometric/datasets/s3dis.py:import h5py
torch_geometric/datasets/s3dis.py:            f = h5py.File(osp.join(self.raw_dir, filename))

```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-torch-geometric/package.py:37]: Ah, sorry, I'm blind. It's even in the `setup.py`, I'm not sure why I didn't notice.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**